// @se/web-ui/collection/components/table-group/table-group.js
import { Component, h, Prop, Element, Host, State } from '@stencil/core';
export class TableGroupComponent {
    constructor() {
        /**
         * Indicates whether or not the `se-table-group` is selectable.  Set to `false` by default.
         */
        this.selectable = false;
        /**
         * Indicates whether or not the `se-table-group` is selected.  Set to `false` by default.
         */
        this.selected = false;
    }
    setButtonId() {
        this.innerId = this.el.getAttribute('id') || '';
    }
    componentWillLoad() {
        this.setButtonId();
    }
    render() {
        return (h(Host, { role: "row", class: { selectable: this.selectable, selected: this.selected } },
            h("div", { class: "table-group-row", id: `wc-${this.innerId}` },
                h("slot", null)),
            h("slot", { name: "detail" })));
    }
    static get is() { return "se-table-group"; }
    static get encapsulation() { return "shadow"; }
    static get originalStyleUrls() { return {
        "$": ["table-group.scss"]
    }; }
    static get styleUrls() { return {
        "$": ["table-group.css"]
    }; }
    static get properties() { return {
        "selectable": {
            "type": "boolean",
            "mutable": false,
            "complexType": {
                "original": "boolean",
                "resolved": "boolean",
                "references": {}
            },
            "required": false,
            "optional": false,
            "docs": {
                "tags": [],
                "text": "Indicates whether or not the `se-table-group` is selectable.  Set to `false` by default."
            },
            "attribute": "selectable",
            "reflect": false,
            "defaultValue": "false"
        },
        "selected": {
            "type": "boolean",
            "mutable": false,
            "complexType": {
                "original": "boolean",
                "resolved": "boolean",
                "references": {}
            },
            "required": false,
            "optional": false,
            "docs": {
                "tags": [],
                "text": "Indicates whether or not the `se-table-group` is selected.  Set to `false` by default."
            },
            "attribute": "selected",
            "reflect": false,
            "defaultValue": "false"
        }
    }; }
    static get states() { return {
        "innerId": {}
    }; }
    static get elementRef() { return "el"; }
}
