// @se/web-ui/esm-es5/appInfo-46118859.js
function getFullTitle(appTitle) {
    var last = '';
    var first = '';
    if (appTitle && appTitle.length) {
        var titleArray = appTitle.split(" ");
        if (titleArray.length > 1) {
            last = titleArray.pop();
        }
        first = titleArray.length ? titleArray.join(" ") : '';
    }
    return {
        first: first,
        last: last
    };
}
function getFullCopyright(copyright) {
    var last = '';
    var first = '';
    if (copyright && copyright.length) {
        var titleArray = copyright.split(" ©");
        if (titleArray.length > 1) {
            last = "©" + titleArray.pop();
        }
        first = titleArray.length ? titleArray.join(" ") : '';
    }
    return {
        first: first,
        last: last
    };
}
var AppInfo = {
    link: "www.schneider-electric.com",
    domain: "ecostruxure",
    copyright: "This application is protected by copyright law and international treaties. © 2020 Schneider Electric Industries SAS. All Rights Reserved."
};
export { AppInfo as A, getFullCopyright as a, getFullTitle as g };
