// @se/web-ui/esm-es5/se-breadcrumb_2.entry.js
import { r as registerInstance, h, g as getElement, H as Host } from './index-8c7316b3.js';
import { a as arrow2_right } from './arrow2_right-4a0b22c1.js';
var breadcrumbCss = ":host{display:-ms-flexbox;display:flex;cursor:pointer}:host ol{margin:0;padding:0}";
var BreadcrumbComponent = /** @class */ (function () {
    function BreadcrumbComponent(hostRef) {
        registerInstance(this, hostRef);
        this.items = [];
    }
    BreadcrumbComponent.prototype.updateLastItem = function () {
        var _this = this;
        this.items = Array.from(this.el.querySelectorAll('se-breadcrumb-item'));
        this.items.forEach(function (item) {
            item.isLast = (item === _this.items[_this.items.length - 1]);
        });
    };
    BreadcrumbComponent.prototype.watchItemList = function () {
        var _this = this;
        this.observer = new MutationObserver(function (mutations) {
            mutations.forEach(function (mutation) {
                if (mutation.addedNodes.length || mutation.removedNodes.length) {
                    _this.updateLastItem();
                }
            });
        });
        this.observer.observe(this.el, { childList: true });
    };
    BreadcrumbComponent.prototype.componentWillLoad = function () {
        this.updateLastItem();
        this.watchItemList();
    };
    BreadcrumbComponent.prototype.componentDidUnload = function () {
        this.observer.disconnect();
    };
    BreadcrumbComponent.prototype.render = function () {
        return (h("nav", { "aria-label": "breadcrumb" }, h("ol", null, h("slot", null))));
    };
    Object.defineProperty(BreadcrumbComponent.prototype, "el", {
        get: function () { return getElement(this); },
        enumerable: true,
        configurable: true
    });
    return BreadcrumbComponent;
}());
BreadcrumbComponent.style = breadcrumbCss;
var breadcrumbItemCss = ":host div{display:-ms-inline-flexbox;display:inline-flex;-ms-flex-align:center;align-items:center}:host .breadcrumb-item{display:inline;color:#333333;font-size:14px;line-height:14px;padding-right:4px;vertical-align:super}:host .breadcrumb-item:hover{text-decoration:underline}:host a{color:inherit;text-decoration:inherit;cursor:inherit}:host se-icon{cursor:default}:host se-icon:active,:host se-icon:focus{color:inherit;background-color:inherit}:host .selected .breadcrumb-item{cursor:default;text-decoration:none;color:#CBCBCB}";
var BreadcrumbItemComponent = /** @class */ (function () {
    function BreadcrumbItemComponent(hostRef) {
        registerInstance(this, hostRef);
        /**
         * Indicates whether or not the breadcrumb item should be selected.  The default setting is `false`.
         */
        this.isLast = false;
    }
    BreadcrumbItemComponent.prototype.render = function () {
        return (h(Host, { role: "listitem" }, h("div", { class: { 'selected': this.isLast } }, this.isLast ?
            h("span", { "aria-current": 'page', class: "breadcrumb-item" }, h("slot", null))
            :
                [h("span", { class: "breadcrumb-item" }, h("a", { href: this.href }, h("slot", null))),
                    h("se-icon", { size: "small" }, h("span", { innerHTML: arrow2_right }))])));
    };
    return BreadcrumbItemComponent;
}());
BreadcrumbItemComponent.style = breadcrumbItemCss;
export { BreadcrumbComponent as se_breadcrumb, BreadcrumbItemComponent as se_breadcrumb_item };
