// @se/web-ui/esm-es5/se-header_3.entry.js
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __generator = (this && this.__generator) || function (thisArg, body) {
    var _ = { label: 0, sent: function() { if (t[0] & 1) throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
    return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function() { return this; }), g;
    function verb(n) { return function (v) { return step([n, v]); }; }
    function step(op) {
        if (f) throw new TypeError("Generator is already executing.");
        while (_) try {
            if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done) return t;
            if (y = 0, t) op = [op[0] & 2, t.value];
            switch (op[0]) {
                case 0: case 1: t = op; break;
                case 4: _.label++; return { value: op[1], done: false };
                case 5: _.label++; y = op[1]; op = [0]; continue;
                case 7: op = _.ops.pop(); _.trys.pop(); continue;
                default:
                    if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }
                    if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }
                    if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }
                    if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }
                    if (t[2]) _.ops.pop();
                    _.trys.pop(); continue;
            }
            op = body.call(thisArg, _);
        } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }
        if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };
    }
};
import { r as registerInstance, h, g as getElement, c as createEvent, H as Host } from './index-8c7316b3.js';
import { A as AppInfo, g as getFullTitle } from './appInfo-46118859.js';
var burgerMenu = "<?xml version=\"1.0\" encoding=\"utf-8\"?>\n<!-- Generator: Adobe Illustrator 23.0.2, SVG Export Plug-In . SVG Version: 6.00 Build 0)  -->\n<svg version=\"1.1\" id=\"Layer_1\" xmlns=\"http://www.w3.org/2000/svg\" xmlns:xlink=\"http://www.w3.org/1999/xlink\" x=\"0px\" y=\"0px\"\n\t viewBox=\"0 0 1024 1024\" style=\"enable-background:new 0 0 1024 1024;\" xml:space=\"preserve\">\n<rect x=\"128\" y=\"468.8\" width=\"768\" height=\"86.4\"/>\n<rect x=\"128\" y=\"211\" width=\"768\" height=\"86\"/>\n<rect x=\"128\" y=\"726.8\" width=\"768\" height=\"86.4\"/>\n</svg>\n";
var headerCss = "@-webkit-keyframes fadeIn{from{opacity:0}to{opacity:1}}@keyframes fadeIn{from{opacity:0}to{opacity:1}}@-webkit-keyframes fadeOut{from{opacity:1}to{opacity:0}}@keyframes fadeOut{from{opacity:1}to{opacity:0}}@-webkit-keyframes slideInLeft{from{-webkit-transform:translate3d(-100%, 0, 0);transform:translate3d(-100%, 0, 0);visibility:visible}to{-webkit-transform:translate3d(0, 0, 0);transform:translate3d(0, 0, 0)}}@keyframes slideInLeft{from{-webkit-transform:translate3d(-100%, 0, 0);transform:translate3d(-100%, 0, 0);visibility:visible}to{-webkit-transform:translate3d(0, 0, 0);transform:translate3d(0, 0, 0)}}@-webkit-keyframes slideOutLeft{from{-webkit-transform:translate3d(0, 0, 0);transform:translate3d(0, 0, 0);visibility:visible}to{-webkit-transform:translate3d(-100%, 0, 0);transform:translate3d(-100%, 0, 0)}}@keyframes slideOutLeft{from{-webkit-transform:translate3d(0, 0, 0);transform:translate3d(0, 0, 0);visibility:visible}to{-webkit-transform:translate3d(-100%, 0, 0);transform:translate3d(-100%, 0, 0)}}@-webkit-keyframes slideCollapseDown{from{-webkit-transform:scaleY(0);transform:scaleY(0);visibility:visible}to{-webkit-transform:scaleY(1);transform:scaleY(1)}}@keyframes slideCollapseDown{from{-webkit-transform:scaleY(0);transform:scaleY(0);visibility:visible}to{-webkit-transform:scaleY(1);transform:scaleY(1)}}@-webkit-keyframes slideCollapseUp{from{-webkit-transform:scaleY(1);transform:scaleY(1)}to{-webkit-transform:scaleY(0);transform:scaleY(0)}}@keyframes slideCollapseUp{from{-webkit-transform:scaleY(1);transform:scaleY(1)}to{-webkit-transform:scaleY(0);transform:scaleY(0)}}:host{-webkit-box-sizing:border-box;box-sizing:border-box;height:auto;padding:0;width:100%;min-height:64px;max-height:64px;z-index:5;display:-ms-flexbox;display:flex;-moz-flex-direction:row;-ms-flex-direction:row;flex-direction:row;-ms-flex-pack:justify;justify-content:space-between;-moz-align-items:center;-ms-flex-align:center;align-items:center;white-space:nowrap;background-color:#FFFFFF;background-color:var(--se-alternative, #FFFFFF)}::slotted([slot=end]){display:-ms-flexbox;display:flex;-ms-flex-align:center;align-items:center}.fill-space{-moz-flex:1 1 auto;-ms-flex:1 1 auto;flex:1 1 auto}.d-flex{-ms-flex-pack:center;justify-content:center;-ms-flex-align:center;align-items:center;display:-ms-flexbox;display:flex;-moz-flex-direction:row;-ms-flex-direction:row;flex-direction:row}.d-flex-column{-moz-align-items:left;-ms-flex-align:left;align-items:left;display:-ms-flexbox;display:flex;-moz-flex-direction:column;-ms-flex-direction:column;flex-direction:column}.header-title-wrapper{margin-right:16px}.project-section{color:#626469;margin:0 16px;font-size:16px;font-weight:300;line-height:24px;text-align:left}.header-menu-button{margin-left:6px}.menu-sidenav{cursor:pointer;margin-right:-17px;padding:8px 16px}.menu-sidenav se-icon{vertical-align:-webkit-baseline-middle;vertical-align:-moz-middle-with-baseline}.header-title{color:#3DCD58;color:var(--se-primary, #3DCD58);padding-left:18px;font-size:20px;line-height:20px;margin:0}.header-title .light{font-weight:300}.header-title-type{color:#3DCD58;color:var(--se-primary, #3DCD58);line-height:24px;padding-left:18px;font-size:12px;margin:1px 0 -4px 0}.mat-toolbar-single-row{padding:0}.font-lighter{font-weight:100}se-icon-schneider{width:140px;padding-left:9px !important;margin-top:5px}se-icon-ecostruxure{margin-bottom:-5px}.padding-container.link-container{font-size:13px}@media (max-width: 599px){se-icon-schneider{display:none}.mat-toolbar-single-row{height:64px}}";
var HeaderComponent = /** @class */ (function () {
    function class_1(hostRef) {
        registerInstance(this, hostRef);
        /**
         * Sets the title of your application.
         */
        this.appTitle = "";
        /**
         * Defines the domain of the application. By default, the domain is `ecostruxure`. If `none`, then no domain will be displayed.
         */
        this.domain = AppInfo.domain;
    }
    class_1.prototype.onClickMenu = function () {
        this.elLeftNav.toggle();
    };
    class_1.prototype.componentWillLoad = function () {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                this.elLeftNav = this.el.querySelector("se-sidemenu");
                this.hasMenu = !!this.elLeftNav;
                return [2 /*return*/];
            });
        });
    };
    class_1.prototype.render = function () {
        var _this = this;
        var title = getFullTitle(this.appTitle);
        var domain;
        if (this.domain.toLowerCase() === "ecostruxure") {
            domain = h("se-icon-ecostruxure", { class: "header-title-type" });
        }
        else if (this.domain.toLowerCase() !== "none") {
            domain = h("span", { class: "header-title-type" }, this.domain);
        }
        return [
            h("div", { class: "d-flex" }, this.hasMenu ?
                h("span", { class: "menu-sidenav", onClick: function () { return _this.onClickMenu(); } }, h("se-icon", { size: "medium", color: "primary" }, h("span", { innerHTML: burgerMenu })))
                : '', h("div", { class: "d-flex-column header-title-wrapper" }, domain, h("h1", { class: "header-title no-margin" }, h("span", null, title.first), h("span", { class: "light" }, "\u00A0", title.last))), this.project ? h("span", { class: "project-section" }, this.project) : ''),
            h("div", { class: "padding-container d-flex" }, h("slot", { name: "start" })),
            h("div", { class: "fill-space center-header-container" }, h("slot", null)),
            h("div", { class: "padding-container d-flex" }, h("slot", { name: "end" }), h("se-icon-schneider", { class: "header-title-type" }))
        ];
    };
    Object.defineProperty(class_1.prototype, "el", {
        get: function () { return getElement(this); },
        enumerable: true,
        configurable: true
    });
    return class_1;
}());
HeaderComponent.style = headerCss;
var test_results_nok = "<svg xmlns=\"http://www.w3.org/2000/svg\" viewBox=\"0 0 1024 1024\"><path d=\"M542 512l134 134c8 9 8 22 0 30-4 5-9 7-15 7-5 0-11-2-15-7L512 542 378 676c-5 5-10 7-16 7-5 0-11-2-15-7-8-8-8-21 0-30l134-134-134-134c-8-9-8-22 0-30 9-9 22-9 31 0l134 134 134-134c8-9 22-9 30 0 8 8 8 21 0 30z\"/></svg>";
var sidemenuCss = "@-webkit-keyframes fadeIn{from{opacity:0}to{opacity:1}}@keyframes fadeIn{from{opacity:0}to{opacity:1}}@-webkit-keyframes fadeOut{from{opacity:1}to{opacity:0}}@keyframes fadeOut{from{opacity:1}to{opacity:0}}@-webkit-keyframes slideInLeft{from{-webkit-transform:translate3d(-100%, 0, 0);transform:translate3d(-100%, 0, 0);visibility:visible}to{-webkit-transform:translate3d(0, 0, 0);transform:translate3d(0, 0, 0)}}@keyframes slideInLeft{from{-webkit-transform:translate3d(-100%, 0, 0);transform:translate3d(-100%, 0, 0);visibility:visible}to{-webkit-transform:translate3d(0, 0, 0);transform:translate3d(0, 0, 0)}}@-webkit-keyframes slideOutLeft{from{-webkit-transform:translate3d(0, 0, 0);transform:translate3d(0, 0, 0);visibility:visible}to{-webkit-transform:translate3d(-100%, 0, 0);transform:translate3d(-100%, 0, 0)}}@keyframes slideOutLeft{from{-webkit-transform:translate3d(0, 0, 0);transform:translate3d(0, 0, 0);visibility:visible}to{-webkit-transform:translate3d(-100%, 0, 0);transform:translate3d(-100%, 0, 0)}}@-webkit-keyframes slideCollapseDown{from{-webkit-transform:scaleY(0);transform:scaleY(0);visibility:visible}to{-webkit-transform:scaleY(1);transform:scaleY(1)}}@keyframes slideCollapseDown{from{-webkit-transform:scaleY(0);transform:scaleY(0);visibility:visible}to{-webkit-transform:scaleY(1);transform:scaleY(1)}}@-webkit-keyframes slideCollapseUp{from{-webkit-transform:scaleY(1);transform:scaleY(1)}to{-webkit-transform:scaleY(0);transform:scaleY(0)}}@keyframes slideCollapseUp{from{-webkit-transform:scaleY(1);transform:scaleY(1)}to{-webkit-transform:scaleY(0);transform:scaleY(0)}}:host{display:none;position:absolute;top:0;left:0;height:100%;width:100%;z-index:160;background-color:transparent !important}:host(.show-menu){display:block}:host(.show-menu) .menu-background{background:rgba(255, 255, 255, 0.6);height:100%;position:absolute;top:0;right:0;bottom:0;left:0;width:100%;z-index:100}:host(.show-menu) .actual-menu{background-color:#FFFFFF;-webkit-box-shadow:rgba(0, 0, 0, 0.3) 1px 1px 4px 0px;box-shadow:rgba(0, 0, 0, 0.3) 1px 1px 4px 0px;height:100vh;max-height:100%;overflow:hidden;position:relative;width:250px;max-width:80%;z-index:150}:host(.open-item) .actual-menu{width:80%}.menu-background.show-menu{-webkit-animation-name:fadeIn;animation-name:fadeIn}.menu-background.hide-menu{-webkit-animation-name:fadeOut;animation-name:fadeOut}.actual-menu.show-menu{-webkit-animation-name:slideInLeft;animation-name:slideInLeft}.actual-menu.hide-menu{-webkit-animation-name:slideOutLeft;animation-name:slideOutLeft}.animated{-webkit-animation-duration:0.2s;animation-duration:0.2s;animation-duration:0.2s;-webkit-animation-fill-mode:both;animation-fill-mode:both}.flex{display:-ms-flexbox;display:flex;-moz-flex:0 1 auto;-ms-flex:0 1 auto;flex:0 1 auto;width:100%}.flex.d-flex-center{-ms-flex-align:center;align-items:center}.menu-sidenav{margin-right:-6px;cursor:pointer}.menu-sidenav se-icon{vertical-align:-webkit-baseline-middle;vertical-align:-moz-middle-with-baseline}.header-title{margin:0}.footer-icon{text-align:center}.footer-icon:last-child{margin-bottom:3px}.external-link{background-color:#F7F7F7;padding:16px;text-align:center}";
var SHOW_MENU = 'show-menu';
var HIDE_MENU = 'hide-menu';
var OPEN_ITEM = 'open-item';
var SidemenuComponent = /** @class */ (function () {
    function class_2(hostRef) {
        registerInstance(this, hostRef);
        this.open = false;
        this.items = [];
        /**
         * Defines the text displayed in the header of the Sidemenu.
         * The default value is `Menu`.
         */
        this.label = 'Menu';
        /*
        * Defines the link to be uses in the external-link element of the Sidemenu.
        * The default value is www.se.com, which will generate if no link is defined.
        * If a different url is provided it will replace the default value.
        * If an empty string is provided the external link element will not be generated.
       */
        this.link = 'www.se.com';
        this.toggled = createEvent(this, "toggled", 7);
    }
    /**
     * Toggle the sidemenu. Optionally, pass the `item` or `id` of a sidemenu-item to open that particular menu item.
     *
     * ex: `document.getElementById("main-sidemenu").toggle("side-about");`
     */
    class_2.prototype.toggle = function (itemName) {
        return __awaiter(this, void 0, void 0, function () {
            var itemElement;
            var _this = this;
            return __generator(this, function (_a) {
                // Only perform toggle if no item name is passed or menu is closed
                if (!this.open || !itemName || itemName && !this.open) {
                    // Only perform open animation if menu is closed
                    if (!this.open) {
                        this.el.classList.add(SHOW_MENU);
                        this.addAnimation(null);
                    }
                    this.open = !this.open;
                }
                else if (this.selectedItem && this.isItemElement(this.selectedItem, itemName)) {
                    // Deselect the active item if a new item name is passed
                    this.unselectAll();
                }
                if (this.open) {
                    if (itemName) {
                        itemElement = this.getItemElement(itemName);
                        if (itemElement) {
                            // Select the new active element
                            this.setActive(itemElement);
                        }
                    }
                    else {
                        try {
                            if (this.items.find(function (x) { return x.classList.contains('active'); })) {
                                this.el.classList.add(OPEN_ITEM);
                            }
                        }
                        catch (error) {
                            console.log(error);
                        }
                    }
                    // Dispatch the 'opened' event
                    this.toggled.emit({ state: 'open' });
                }
                else {
                    // Remove css classes and unselect the active element
                    this.removeAnimation(function () {
                        _this.el.classList.remove(SHOW_MENU);
                    });
                    this.unselectAll();
                    // Dispatch the 'closed' event
                    this.toggled.emit({ state: 'closed' });
                }
                return [2 /*return*/];
            });
        });
    };
    class_2.prototype.noSelectedItem = function () {
        var _this = this;
        return !this.items.find(function (x) { return x === _this.selectedItem; });
    };
    class_2.prototype.unselectAll = function () {
        this.el.classList.remove(OPEN_ITEM);
        if (this.selectedItem && !this.selectedItem.childElementCount) {
            this.selectedItem = undefined;
            this.items.forEach(function (item) {
                item.active = false;
            });
        }
    };
    class_2.prototype.getItemElement = function (name) {
        var _this = this;
        return this.items.find(function (i) { return _this.isItemElement(i, name); });
    };
    class_2.prototype.isItemElement = function (elm, name) {
        return elm.getAttribute('item') === name || elm.getAttribute('id') === name;
    };
    class_2.prototype.setActive = function (item) {
        var _this = this;
        if (this.items.length) {
            this.items.forEach(function (item) {
                item.active = false;
            });
            setTimeout(function () {
                item.active = true;
                _this.selectedItem = item;
                _this.el.classList.add(OPEN_ITEM);
            }, 100);
        }
    };
    class_2.prototype.addAnimation = function (callback) {
        var _this = this;
        try {
            this.menuInnerEl.classList.add(SHOW_MENU);
            this.backdropEl.classList.add(SHOW_MENU);
            setTimeout(function () {
                _this.menuInnerEl.classList.remove(SHOW_MENU);
                _this.backdropEl.classList.remove(SHOW_MENU);
                callback && callback();
            }, 200);
        }
        catch (error) {
            console.log(error);
        }
    };
    class_2.prototype.removeAnimation = function (callback) {
        var _this = this;
        try {
            this.menuInnerEl.classList.add(HIDE_MENU);
            this.backdropEl.classList.add(HIDE_MENU);
            setTimeout(function () {
                _this.menuInnerEl.classList.remove(HIDE_MENU);
                _this.backdropEl.classList.remove(HIDE_MENU);
                callback && callback();
            }, 200);
        }
        catch (error) {
            console.log(error);
        }
    };
    class_2.prototype.watchItemList = function () {
        var _this = this;
        this.observer = new MutationObserver(function (mutations) {
            var activeItem = false;
            mutations.forEach(function (mutation) {
                if (mutation.addedNodes.length) {
                    _this.setItemsArray();
                }
                if (mutation.removedNodes.length) {
                    _this.setItemsArray();
                    _this.items.forEach(function (item) {
                        if (item.active) {
                            activeItem = true;
                        }
                    });
                    if (!activeItem && _this.selectedItem) {
                        // console.log(this.selectedItem);
                        _this.selectedItem = undefined;
                    }
                }
            });
        });
        this.observer.observe(this.el, { childList: true });
    };
    class_2.prototype.setItemsArray = function () {
        this.items = Array.from(this.el.querySelectorAll('se-sidemenu-item'));
    };
    class_2.prototype.renderList = function () {
        var _this = this;
        return this.items.map(function (item) {
            return [
                h("se-list-item", { class: [!item.childElementCount ? 'hide-nav-icon' : '', 'sidemenu-list-item'].join(' '), option: "nav", onClick: function () { return _this.setActive(item); }, selected: item.active, item: item.item, id: item.id ? "list-" + item.id : '' }),
            ];
        });
    };
    class_2.prototype.componentWillLoad = function () {
        this.setItemsArray();
        this.watchItemList();
    };
    class_2.prototype.componentDidUnload = function () {
        this.items.length = 0;
        this.selectedItem = undefined;
        this.observer.disconnect();
    };
    class_2.prototype.render = function () {
        var _this = this;
        return [
            h("div", { class: "menu-background animated d-flex-row flex", onClick: function () { return _this.toggle(); }, ref: function (el) { return _this.backdropEl = el; } }),
            h("div", { class: "actual-menu animated full-content d-flex-column flex", ref: function (el) { return _this.menuInnerEl = el; } }, h("div", { class: "d-flex flex" }, h("se-block", { width: "250px" }, h("div", { class: "d-flex-center flex" }, h("span", { class: "menu-sidenav", onClick: function () { return _this.toggle(); } }, h("se-icon", { size: "large", color: "primary" }, h("span", { innerHTML: test_results_nok }))), h("h3", { class: "header-title" }, this.label)), h("se-divider", null), h("se-block-content", { option: "fill" }, h("se-list", { option: "nav" }, this.renderList())), h("se-icon-lifeison", { class: "footer-icon", color: "standard" }), this.link ?
                h("div", { class: "external-link" }, h("se-link", { class: "sidemenu-link", url: "http://" + this.link }, this.link)) : ''), h("se-divider", { option: "vertical" }), h("se-block", { ref: function (el) { return _this.menuItemInnerEl = el; } }, h("slot", null))))
        ];
    };
    Object.defineProperty(class_2.prototype, "el", {
        get: function () { return getElement(this); },
        enumerable: true,
        configurable: true
    });
    return class_2;
}());
SidemenuComponent.style = sidemenuCss;
var sidemenuItemCss = "@-webkit-keyframes fadeIn{from{opacity:0}to{opacity:1}}@keyframes fadeIn{from{opacity:0}to{opacity:1}}@-webkit-keyframes fadeOut{from{opacity:1}to{opacity:0}}@keyframes fadeOut{from{opacity:1}to{opacity:0}}@-webkit-keyframes slideInLeft{from{-webkit-transform:translate3d(-100%, 0, 0);transform:translate3d(-100%, 0, 0);visibility:visible}to{-webkit-transform:translate3d(0, 0, 0);transform:translate3d(0, 0, 0)}}@keyframes slideInLeft{from{-webkit-transform:translate3d(-100%, 0, 0);transform:translate3d(-100%, 0, 0);visibility:visible}to{-webkit-transform:translate3d(0, 0, 0);transform:translate3d(0, 0, 0)}}@-webkit-keyframes slideOutLeft{from{-webkit-transform:translate3d(0, 0, 0);transform:translate3d(0, 0, 0);visibility:visible}to{-webkit-transform:translate3d(-100%, 0, 0);transform:translate3d(-100%, 0, 0)}}@keyframes slideOutLeft{from{-webkit-transform:translate3d(0, 0, 0);transform:translate3d(0, 0, 0);visibility:visible}to{-webkit-transform:translate3d(-100%, 0, 0);transform:translate3d(-100%, 0, 0)}}@-webkit-keyframes slideCollapseDown{from{-webkit-transform:scaleY(0);transform:scaleY(0);visibility:visible}to{-webkit-transform:scaleY(1);transform:scaleY(1)}}@keyframes slideCollapseDown{from{-webkit-transform:scaleY(0);transform:scaleY(0);visibility:visible}to{-webkit-transform:scaleY(1);transform:scaleY(1)}}@-webkit-keyframes slideCollapseUp{from{-webkit-transform:scaleY(1);transform:scaleY(1)}to{-webkit-transform:scaleY(0);transform:scaleY(0)}}@keyframes slideCollapseUp{from{-webkit-transform:scaleY(1);transform:scaleY(1)}to{-webkit-transform:scaleY(0);transform:scaleY(0)}}:host{height:100%;width:100%}:host [role=sidenavpanel]{height:100%}:host(.navitem-hidden){display:none !important;height:0}";
var SidemenuItemComponent = /** @class */ (function () {
    function SidemenuItemComponent(hostRef) {
        registerInstance(this, hostRef);
        /**
         * Defines if the menu item is active or not.
         * The default setting is `false`.
         */
        this.active = false;
        this.didClick = createEvent(this, "didClick", 7);
    }
    SidemenuItemComponent.prototype.activeDidChange = function () {
        if (this.active) {
            this.didClick.emit();
        }
    };
    SidemenuItemComponent.prototype.render = function () {
        return (h(Host, { role: 'sidenavpanel', "aria-hidden": { 'true': !this.active }, class: [this.active ? 'active' : 'navitem-hidden'].join(' ') }, h("slot", null)));
    };
    Object.defineProperty(SidemenuItemComponent, "watchers", {
        get: function () {
            return {
                "active": ["activeDidChange"]
            };
        },
        enumerable: true,
        configurable: true
    });
    return SidemenuItemComponent;
}());
SidemenuItemComponent.style = sidemenuItemCss;
export { HeaderComponent as se_header, SidemenuComponent as se_sidemenu, SidemenuItemComponent as se_sidemenu_item };
