// @se/web-ui/esm-es5/se-visual-linear.entry.js
import { r as registerInstance, h, H as Host } from './index-8c7316b3.js';
var visualLinearCss = ":host{margin:4px}.se-visual-linear{display:-ms-flexbox;display:flex}svg{margin-top:4px;margin-bottom:4px}@media all and (-webkit-min-device-pixel-ratio: 0) and (min-resolution: 0.001dpcm){svg{width:100%}}@-moz-document url-prefix(){svg{width:100%}}@supports (-ms-ime-align: auto){svg{width:80%}}.progress-bar-wrapper{fill:#F7F7F7;width:100%}.progress-bar{color:inherit}text{color:#333333}.linear-value{font-weight:600}.linear-label{font-size:12px}.linear-value+.linear-label{font-size:14px}.inline.se-visual-linear{-ms-flex-direction:row;flex-direction:row;-ms-flex-pack:center;justify-content:center;-moz-align-items:start;-ms-flex-align:start;align-items:start}.inline svg{margin-right:16px}.inline .linear-value{margin-right:4px}.stacked.se-visual-linear{-ms-flex-direction:column-reverse;flex-direction:column-reverse}.stacked .linear-value+.linear-label{float:right}";
var VisualLinearComponent = /** @class */ (function () {
    function VisualLinearComponent(hostRef) {
        registerInstance(this, hostRef);
        /**
         * Defines the layout of your component.
         * `inline` is the default option.  This sets the "progress bar" adjacent to the label.
         * `stacked` option will render the "progress bar" below the label.
         */
        this.option = 'inline';
        /**
         * Sets the color of the visual "progress bar".
         * The string should be a 6-digit hexadecimal color with a hashtag (example: #3dcd58).
         * By default, the progress bar will have a green color.
         */
        this.secolor = 'primary';
        this.isHexColor = false;
    }
    VisualLinearComponent.prototype.colorDidChange = function () {
        this.isHexColor = (this.secolor && this.secolor.indexOf('#') !== -1) ? true : false;
    };
    VisualLinearComponent.prototype.componentDidLoad = function () {
        this.colorDidChange();
    };
    VisualLinearComponent.prototype.render = function () {
        return (h(Host, { class: !this.isHexColor ? "color-" + this.secolor : '' }, h("div", { class: ["se-visual-linear", this.option].join(' ') }, h("svg", { height: "8" }, h("rect", { class: "progress-bar-wrapper", width: "100%", height: "8" }), h("rect", { class: "progress-bar", width: this.percentage + '%', height: "8", style: { fill: this.isHexColor ? this.secolor : 'currentColor' } }), "Sorry, your browser does not support inline SVG."), h("text", null, this.value ? h("tspan", { class: "linear-value" }, this.value) : '', h("tspan", { x: "100%", class: "linear-label" }, this.label)))));
    };
    Object.defineProperty(VisualLinearComponent, "watchers", {
        get: function () {
            return {
                "secolor": ["colorDidChange"]
            };
        },
        enumerable: true,
        configurable: true
    });
    return VisualLinearComponent;
}());
VisualLinearComponent.style = visualLinearCss;
export { VisualLinearComponent as se_visual_linear };
