// @se/web-ui/collection/components/authentication/authentication.js
import { Component, h, Prop } from "@stencil/core";
import { getFullTitle } from "../../utils/utils";
import { AppInfo } from "../appInfo";
export class AuthenticationComponent {
    constructor() {
        /**
         * The domain you want to display.  If set to `ecostruxure`, it renders an EcoStruxure icon.
         */
        this.domain = AppInfo.domain;
        /**
         * The copyright you would like to display.
         */
        this.copyright = AppInfo.copyright;
        /**
         * If set to `true`, this will hide the authentication screen.
         */
        this.hide = false;
    }
    renderAuthentication() {
        const title = getFullTitle(this.appTitle);
        let domain;
        if (this.domain.toLowerCase() === `ecostruxure`) {
            domain = h("se-icon-ecostruxure", { size: "medium" });
        }
        else if (this.domain.toLowerCase() !== 'none') {
            domain = h("span", { class: "header-title-type" }, this.domain);
        }
        return (h("div", { class: "content-wrapper" },
            h("div", { class: "image-container" },
                h("img", { class: "image-background", style: { 'backgroundImage': this.imageUrl } })),
            h("main", null,
                h("div", { class: "green-bars" },
                    h("div", { class: "dark-green-bar" }),
                    h("div", { class: "light-green-bar" }))),
            h("div", { class: "login-container" },
                h("div", { class: "information" },
                    domain,
                    h("h1", { class: "header-title" },
                        h("span", null, title.first),
                        h("span", { class: "light" },
                            "\u00A0",
                            title.last)),
                    h("p", { class: "version" },
                        "version ",
                        this.version)),
                h("div", { class: "form" },
                    h("slot", null))),
            h("footer", null,
                h("div", { class: "copyright-note" },
                    h("span", null, this.copyright)),
                h("div", { class: "footer-logo align-end" },
                    h("se-icon-lifeison", null)))));
    }
    render() {
        return (h("div", { class: ['se-authentication', this.hide ? 'hide' : ''].join(' ') }, !this.hide ? this.renderAuthentication() : ''));
    }
    static get is() { return "se-authentication"; }
    static get encapsulation() { return "shadow"; }
    static get originalStyleUrls() { return {
        "$": ["authentication.scss"]
    }; }
    static get styleUrls() { return {
        "$": ["authentication.css"]
    }; }
    static get properties() { return {
        "appTitle": {
            "type": "string",
            "mutable": false,
            "complexType": {
                "original": "string",
                "resolved": "string",
                "references": {}
            },
            "required": false,
            "optional": false,
            "docs": {
                "tags": [],
                "text": "The title of your authentication screen."
            },
            "attribute": "app-title",
            "reflect": false
        },
        "version": {
            "type": "string",
            "mutable": false,
            "complexType": {
                "original": "string",
                "resolved": "string",
                "references": {}
            },
            "required": false,
            "optional": false,
            "docs": {
                "tags": [],
                "text": "The version number you want to display."
            },
            "attribute": "version",
            "reflect": false
        },
        "domain": {
            "type": "string",
            "mutable": false,
            "complexType": {
                "original": "string",
                "resolved": "string",
                "references": {}
            },
            "required": false,
            "optional": false,
            "docs": {
                "tags": [],
                "text": "The domain you want to display.  If set to `ecostruxure`, it renders an EcoStruxure icon."
            },
            "attribute": "domain",
            "reflect": false,
            "defaultValue": "AppInfo.domain"
        },
        "copyright": {
            "type": "string",
            "mutable": false,
            "complexType": {
                "original": "string",
                "resolved": "string",
                "references": {}
            },
            "required": false,
            "optional": false,
            "docs": {
                "tags": [],
                "text": "The copyright you would like to display."
            },
            "attribute": "copyright",
            "reflect": false,
            "defaultValue": "AppInfo.copyright"
        },
        "imageUrl": {
            "type": "string",
            "mutable": false,
            "complexType": {
                "original": "string",
                "resolved": "string",
                "references": {}
            },
            "required": false,
            "optional": false,
            "docs": {
                "tags": [],
                "text": "Sets the background image for your authentication screen."
            },
            "attribute": "image-url",
            "reflect": false
        },
        "hide": {
            "type": "boolean",
            "mutable": false,
            "complexType": {
                "original": "boolean",
                "resolved": "boolean",
                "references": {}
            },
            "required": false,
            "optional": false,
            "docs": {
                "tags": [],
                "text": "If set to `true`, this will hide the authentication screen."
            },
            "attribute": "hide",
            "reflect": false,
            "defaultValue": "false"
        }
    }; }
}
