// @se/web-ui/esm/appInfo-46118859.js
function getFullTitle(appTitle) {
    let last = '';
    let first = '';
    if (appTitle && appTitle.length) {
        const titleArray = appTitle.split(" ");
        if (titleArray.length > 1) {
            last = titleArray.pop();
        }
        first = titleArray.length ? titleArray.join(" ") : '';
    }
    return {
        first,
        last
    };
}
function getFullCopyright(copyright) {
    let last = '';
    let first = '';
    if (copyright && copyright.length) {
        const titleArray = copyright.split(" ©");
        if (titleArray.length > 1) {
            last = "©" + titleArray.pop();
        }
        first = titleArray.length ? titleArray.join(" ") : '';
    }
    return {
        first,
        last
    };
}

const AppInfo = {
    link: "www.schneider-electric.com",
    domain: "ecostruxure",
    copyright: "This application is protected by copyright law and international treaties. © 2020 Schneider Electric Industries SAS. All Rights Reserved."
};

export { AppInfo as A, getFullCopyright as a, getFullTitle as g };
