// @se/web-ui/esm/se-list_3.entry.js
import { r as registerInstance, h, H as Host, g as getElement, c as createEvent } from './index-8c7316b3.js';
import { a as arrow2_right } from './arrow2_right-4a0b22c1.js';

const listCss = ":host{width:100%;height:100%;position:relative;overflow:auto;-moz-flex-direction:column;-ms-flex-direction:column;flex-direction:column}";

const ListComponent = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        /**
         * Defines the style of the list.  The default setting is `classic`.
         */
        this.option = "classic";
        /**
         * Defines if list groups can be collapsed.  The default setting is `true`.
         */
        this.canCollapse = true;
    }
    optionDidChange() {
        Array.from(this.el.querySelectorAll('se-list-item, se-list-group')).forEach((item) => {
            const closeList = item.closest('se-list');
            // Make sure we only change the style of the current list. Handy if list has a dropdown with a different list style
            if (closeList === this.el) {
                item.option = this.option;
            }
        });
    }
    canCollapseDidChange() {
        Array.from(this.el.querySelectorAll('se-list-group')).forEach((item) => {
            item.canCollapse = this.canCollapse;
        });
    }
    componentWillLoad() {
        this.optionDidChange();
        this.canCollapseDidChange();
    }
    render() {
        return h(Host, { role: "list" }, h("slot", null));
    }
    get el() { return getElement(this); }
    static get watchers() { return {
        "option": ["optionDidChange"],
        "canCollapse": ["canCollapseDidChange"]
    }; }
};
ListComponent.style = listCss;

const arrow2Down = `<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 1024 1024"><path d="M657 432c8 8 8 22 0 30L527 592c-5 4-10 6-15 6-6 0-11-2-16-6L366 462c-8-8-8-22 0-30 9-8 22-8 30 0l116 115 115-115c8-8 21-8 30 0z"/></svg>`;

const arrow2Up = `<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 1024 1024"><path d="M657 562c8 8 8 22 0 30-4 4-10 6-15 6-6 0-11-2-15-6L512 477 396 592c-4 4-9 6-15 6-5 0-11-2-15-6-8-8-8-22 0-30l130-130c9-8 22-8 31 0z"/></svg>`;

const listGroupCss = ":host{margin:0;font-weight:lighter;display:-ms-flexbox;display:flex}:host(.hide-nav-icon) se-icon{display:none}:host(.sidemenu-list-item) .nav-content{font-weight:500}:host(.sidemenu-list-item) .button{border-bottom:1px solid #E6E6E6}.se-list-item{width:100%;width:-webkit-fill-available;-webkit-transition:0.2s;transition:0.2s}.se-list-item.button.classic,.se-list-item.button.nav{min-height:47px;height:20px;max-width:100%}.se-list-item.classic.selected,.se-list-item.dropdown.selected{color:#FFFFFF;color:var(--se-selected-contrast, #FFFFFF);background-color:#626469;background-color:var(--se-selected, #626469)}.se-list-item.classic.selected:hover,.se-list-item.classic.selected:focus,.se-list-item.dropdown.selected:hover,.se-list-item.dropdown.selected:focus{color:#0F0F0F;color:var(--se-hover-standard-contrast, #0F0F0F);background-color:#CBCBCB;background-color:var(--se-hover-standard, #CBCBCB)}.se-list-item.nav.selected{color:#3DCD58;color:var(--se-primary, #3DCD58)}.se-list-item.treeview{border-bottom:0;padding:4px 0}.se-list-item.treeview .nav-content{padding-left:12px}.se-list-item.treeview .nav-icon{margin-left:2px;margin-right:-8px}.se-list-item.treeview.selected{color:#FFFFFF;color:var(--se-selected-contrast, #FFFFFF);background-color:#626469;background-color:var(--se-selected, #626469)}.se-list-item.treeview.selected:hover,.se-list-item.treeview.selected:focus{color:#0F0F0F;color:var(--se-hover-standard-contrast, #0F0F0F);background-color:#CBCBCB;background-color:var(--se-hover-standard, #CBCBCB)}.se-list-item.dropdown{min-height:32px;max-width:100%;border-bottom:none;padding-left:0}.se-list-item.dropdown .nav-content{padding:6px 10px}.se-list-item.dropdown .nav-content .list-item-label{font-size:16px;line-height:20px}.se-list-item.dropdown .nav-icon{margin-right:0;margin-left:4px}.se-list-item.headline{min-height:42px;height:42px}.se-list-item.headline .nav-content small{padding-bottom:2px}.button{-webkit-box-sizing:border-box;box-sizing:border-box;border:none;margin:0;padding:0;overflow:visible;position:relative;outline:none;background:transparent;width:100%;color:inherit;font:inherit;display:-ms-flexbox;display:flex;-moz-align-items:center;-ms-flex-align:center;align-items:center;-moz-flex:1 1 auto;-ms-flex:1 1 auto;flex:1 1 auto;text-align:left;text-align:start;border-bottom:1px solid #E6E6E6;text-decoration:none}.button:not([disabled]){cursor:pointer}.button:not([disabled]):hover,.button:not([disabled]):focus{background-color:#EDEDED}.selectedBar{position:absolute;top:0;bottom:0;left:0;width:5px;background-color:#3DCD58;background-color:var(--se-primary, #3DCD58)}.nav-icon{margin:-1px -12px 0 8px}.nav-content{min-width:0;padding:0 16px;margin:0;overflow:hidden;font-weight:300;font-size:16px;-moz-flex:1 1 auto;-ms-flex:1 1 auto;flex:1 1 auto;display:-ms-flexbox;display:flex;-moz-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-moz-align-items:start center;-ms-flex-align:start center;align-items:start center}.nav-content div,.nav-content small{white-space:nowrap;overflow:hidden;text-overflow:ellipsis}.nav-content .list-item-label{line-height:24px}.nav-content small{font-size:14px;line-height:14px}se-list-item .nav-icon{margin-left:18px}:host{-moz-flex-direction:column;-ms-flex-direction:column;flex-direction:column}.group-item{overflow:unset;padding-bottom:0}.se-list-group{list-style-type:none;margin:0;padding:0}.se-list-group.button .list-group-label{line-height:24px}.se-list-group.button.collapsed+.group-item{display:none}.se-list-group.button.selected{color:#FFFFFF;color:var(--se-selected-contrast, #FFFFFF);background-color:#626469;background-color:var(--se-selected, #626469)}.se-list-group.button.selected:hover,.se-list-group.button.selected:focus{background-color:#434343}.se-list-group.button.selectedChild{color:#FFFFFF;color:var(--se-selected-contrast, #FFFFFF);background-color:#626469;background-color:var(--se-selected, #626469)}.se-list-group.button.selectedChild:hover,.se-list-group.button.selectedChild:focus{background-color:#434343;background-color:var(--se-focus, #434343)}.se-list-group.button.classic,.se-list-group.button.nav{min-height:47px;height:20px;max-width:100%}.se-list-group.button.treeview{border-bottom:0;padding:4px 0}.se-list-group.button.treeview>*:first-child{padding-left:4px}.se-list-group.button.treeview .nav-icon{margin-right:0}.se-list-group.button.treeview .nav-content{padding-left:8px}.se-list-group.button.treeview .treeview-collapse-icon{margin-left:2px}.se-list-group.button.treeview+div se-list-item{padding-left:6px}.se-list-group.button.headline{height:47px;background-color:#F7F7F7}.se-list-group.button.headline .nav-content{font-size:16px}.se-list-group.button.dropdown{margin-top:-1px}";

const ListGroupComponent = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        /**
         * Defines if the item group is collapsed/closed. The default setting is `false`.
         */
        this.collapsed = false;
        /**
         * Defines the group indentation to add paddings to the list item (used with multiple list groups).
         */
        this.indentation = 0;
        /**
         * Defines if list groups can be collapsed, true by default.
         */
        this.canCollapse = true;
        this.didGroupClick = createEvent(this, "didGroupClick", 7);
        this.didGroupCollapse = createEvent(this, "didGroupCollapse", 7);
    }
    collapsedChanged() {
        this.checkSelected();
    }
    ChildUpdated() {
        this.checkSelected();
    }
    checkSelected() {
        if (!this.collapsed) {
            this.selectedChild = false;
        }
        else {
            Array.from(this.el.querySelectorAll("se-list-item, se-list-group")).forEach((item) => {
                if (item.selected || item.selectedChild) {
                    this.selectedChild = true;
                }
            });
        }
    }
    toggleGroupButton() {
        if (this.option !== 'treeview') {
            this.collapsed = !this.collapsed;
            this.didGroupCollapse.emit({ collapsed: this.collapsed });
        }
        this.didGroupClick.emit();
    }
    toggleCollapseTreeview() {
        // called only from treeview
        this.collapsed = !this.collapsed;
        this.didGroupCollapse.emit({ collapsed: this.collapsed });
    }
    setButtonId() {
        const id = this.el.getAttribute('id');
        if (id) {
            this.innerId = `wc-${id}`;
        }
    }
    getClosestParent() {
        // get the closest between se-list or se-list-group
        return this.el.parentElement.closest("se-list, se-list-group");
    }
    getParentConfig() {
        const closest = this.getClosestParent() || {};
        const indentation = closest.indentation;
        if (indentation !== null && indentation !== undefined) {
            this.indentation = indentation + 1;
        }
        if (closest.option) {
            this.option = closest.option;
        }
    }
    componentWillLoad() {
        this.getParentConfig();
        this.hasIcon = !!this.icon || !!this.el.querySelector('[slot="icon"]');
    }
    componentDidLoad() {
        this.setButtonId();
    }
    render() {
        let title = this.item;
        let myDescription = null;
        if (!!this.description) {
            myDescription = h("small", null, this.description);
            title = `${title}, ${this.description}`;
        }
        // The button section is a copy of the list item. External component cannot be used inside a component (DOM issue)
        return (h(Host, { role: "listitem", option: this.option }, h("button", { "aria-expanded": `${this.collapsed}`, title: title, id: this.innerId, class: ['se-list-group', this.option, this.collapsed ? "collapsed" : '', this.option, this.selected ? "selected" : '', this.selectedChild ? "selectedChild" : '', "button"].join(' '), style: { paddingLeft: `${20 * this.indentation}px` }, onClick: () => this.toggleGroupButton(), disabled: !this.canCollapse }, this.option === "nav" && this.selected && h("div", { class: "selectedBar" }), this.option === 'treeview' ? h("se-icon", { class: "treeview-collapse-icon", onClick: (e) => { e.stopPropagation(); this.toggleCollapseTreeview(); }, style: { paddingLeft: `calc(${8 * this.indentation}px)` } }, h("span", { innerHTML: this.collapsed ? arrow2_right : arrow2Down })) : '', this.hasIcon &&
            h("div", { class: "nav-icon" }, this.icon && h("se-icon", { color: this.iconColor }, this.icon), h("slot", { name: "icon" })), h("div", { class: "nav-content" }, h("div", { class: "list-group-label" }, this.item), myDescription), this.option !== "treeview" && this.canCollapse
            ? h("se-icon", { class: "standard-collapse-icon", size: "medium" }, h("span", { innerHTML: this.collapsed ? arrow2Down : arrow2Up }))
            : ''), h("div", { role: "list", class: ["se-list-group", "group-item", this.option].join(' ') }, h("slot", null))));
    }
    get el() { return getElement(this); }
    static get watchers() { return {
        "collapsed": ["collapsedChanged"]
    }; }
};
ListGroupComponent.style = listGroupCss;

const listItemCss = ":host{margin:0;font-weight:lighter;display:-ms-flexbox;display:flex}:host(.hide-nav-icon) se-icon{display:none}:host(.sidemenu-list-item) .nav-content{font-weight:500}:host(.sidemenu-list-item) .button{border-bottom:1px solid #E6E6E6}.se-list-item{width:100%;width:-webkit-fill-available;-webkit-transition:0.2s;transition:0.2s}.se-list-item.button.classic,.se-list-item.button.nav{min-height:47px;height:20px;max-width:100%}.se-list-item.classic.selected,.se-list-item.dropdown.selected{color:#FFFFFF;color:var(--se-selected-contrast, #FFFFFF);background-color:#626469;background-color:var(--se-selected, #626469)}.se-list-item.classic.selected:hover,.se-list-item.classic.selected:focus,.se-list-item.dropdown.selected:hover,.se-list-item.dropdown.selected:focus{color:#0F0F0F;color:var(--se-hover-standard-contrast, #0F0F0F);background-color:#CBCBCB;background-color:var(--se-hover-standard, #CBCBCB)}.se-list-item.nav.selected{color:#3DCD58;color:var(--se-primary, #3DCD58)}.se-list-item.treeview{border-bottom:0;padding:4px 0}.se-list-item.treeview .nav-content{padding-left:12px}.se-list-item.treeview .nav-icon{margin-left:2px;margin-right:-8px}.se-list-item.treeview.selected{color:#FFFFFF;color:var(--se-selected-contrast, #FFFFFF);background-color:#626469;background-color:var(--se-selected, #626469)}.se-list-item.treeview.selected:hover,.se-list-item.treeview.selected:focus{color:#0F0F0F;color:var(--se-hover-standard-contrast, #0F0F0F);background-color:#CBCBCB;background-color:var(--se-hover-standard, #CBCBCB)}.se-list-item.dropdown{min-height:32px;max-width:100%;border-bottom:none;padding-left:0}.se-list-item.dropdown .nav-content{padding:6px 10px}.se-list-item.dropdown .nav-content .list-item-label{font-size:16px;line-height:20px}.se-list-item.dropdown .nav-icon{margin-right:0;margin-left:4px}.se-list-item.headline{min-height:42px;height:42px}.se-list-item.headline .nav-content small{padding-bottom:2px}.button{-webkit-box-sizing:border-box;box-sizing:border-box;border:none;margin:0;padding:0;overflow:visible;position:relative;outline:none;background:transparent;width:100%;color:inherit;font:inherit;display:-ms-flexbox;display:flex;-moz-align-items:center;-ms-flex-align:center;align-items:center;-moz-flex:1 1 auto;-ms-flex:1 1 auto;flex:1 1 auto;text-align:left;text-align:start;border-bottom:1px solid #E6E6E6;text-decoration:none}.button:not([disabled]){cursor:pointer}.button:not([disabled]):hover,.button:not([disabled]):focus{background-color:#EDEDED}.selectedBar{position:absolute;top:0;bottom:0;left:0;width:5px;background-color:#3DCD58;background-color:var(--se-primary, #3DCD58)}.nav-icon{margin:-1px -12px 0 8px}.nav-content{min-width:0;padding:0 16px;margin:0;overflow:hidden;font-weight:300;font-size:16px;-moz-flex:1 1 auto;-ms-flex:1 1 auto;flex:1 1 auto;display:-ms-flexbox;display:flex;-moz-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-moz-align-items:start center;-ms-flex-align:start center;align-items:start center}.nav-content div,.nav-content small{white-space:nowrap;overflow:hidden;text-overflow:ellipsis}.nav-content .list-item-label{line-height:24px}.nav-content small{font-size:14px;line-height:14px}se-list-item .nav-icon{margin-left:18px}";

const ListItemComponent = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        /**
         * Defines the group indentation to add paddings to the list item (used with multiple list groups).
         */
        this.indentation = 0;
        this.didSelectedChange = createEvent(this, "didSelectedChange", 7);
    }
    SelectedDidChange() {
        this.didSelectedChange.emit();
    }
    setButtonId() {
        const id = this.el.getAttribute('id');
        if (id) {
            this.innerId = `wc-${id}`;
        }
    }
    getClosestParent() {
        // get the closest between se-list or se-list-group
        return this.el.parentElement.closest("se-list, se-list-group");
    }
    getParentConfig() {
        const closest = this.getClosestParent() || {};
        const indentation = closest.indentation;
        if (indentation !== null && indentation !== undefined) {
            this.indentation = indentation + 1;
        }
        if (closest.option) {
            this.option = closest.option;
        }
    }
    componentWillLoad() {
        this.getParentConfig();
    }
    componentDidLoad() {
        this.setButtonId();
    }
    render() {
        let myDescription = null;
        let title = this.item;
        if (!!this.description) {
            myDescription = h("small", null, this.description);
            title = `${title}, ${this.description}`;
        }
        const TagType = this.href === undefined ? 'button' : 'a';
        const attrs = (TagType === 'a') ? { href: this.href } : {};
        const icon = this.option !== "treeview" ?
            h("se-icon", { color: this.iconColor }, this.icon)
            :
                this.indentation !== 0 ?
                    h("se-icon", { color: this.iconColor, style: { paddingLeft: `8px` } }, this.icon)
                    :
                        h("se-icon", { color: this.iconColor }, this.icon);
        const padding = this.option !== "treeview" ? 20 : 24;
        return (h(Host, { role: "listitem" }, h(TagType, Object.assign({}, attrs, { title: title, class: { "selected": this.selected, ["button"]: true, [this.option]: true, ["se-list-item"]: true }, style: { paddingLeft: `${padding * this.indentation}px` }, id: this.innerId }), (this.option === "nav" && this.selected) ? h("div", { class: "selectedBar" }) : '', h("div", { class: "nav-icon" }, !!this.icon ? icon : '', h("slot", { name: "icon" })), h("div", { class: "nav-content" }, h("div", { class: "list-item-label" }, this.item), myDescription), h("slot", null), this.option === "nav" ? h("se-icon", { size: "medium" }, h("span", { innerHTML: arrow2_right })) : '')));
    }
    get el() { return getElement(this); }
    static get watchers() { return {
        "selected": ["SelectedDidChange"]
    }; }
};
ListItemComponent.style = listItemCss;

export { ListComponent as se_list, ListGroupComponent as se_list_group, ListItemComponent as se_list_item };
