// @se/web-ui/cjs/se-visual-linear.cjs.entry.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const index = require('./index-fec6655d.js');

const visualLinearCss = ":host{margin:4px}.se-visual-linear{display:-ms-flexbox;display:flex}svg{margin-top:4px;margin-bottom:4px}@media all and (-webkit-min-device-pixel-ratio: 0) and (min-resolution: 0.001dpcm){svg{width:100%}}@-moz-document url-prefix(){svg{width:100%}}@supports (-ms-ime-align: auto){svg{width:80%}}.progress-bar-wrapper{fill:#F7F7F7;width:100%}.progress-bar{color:inherit}text{color:#333333}.linear-value{font-weight:600}.linear-label{font-size:12px}.linear-value+.linear-label{font-size:14px}.inline.se-visual-linear{-ms-flex-direction:row;flex-direction:row;-ms-flex-pack:center;justify-content:center;-moz-align-items:start;-ms-flex-align:start;align-items:start}.inline svg{margin-right:16px}.inline .linear-value{margin-right:4px}.stacked.se-visual-linear{-ms-flex-direction:column-reverse;flex-direction:column-reverse}.stacked .linear-value+.linear-label{float:right}";

const VisualLinearComponent = class {
    constructor(hostRef) {
        index.registerInstance(this, hostRef);
        /**
         * Defines the layout of your component.
         * `inline` is the default option.  This sets the "progress bar" adjacent to the label.
         * `stacked` option will render the "progress bar" below the label.
         */
        this.option = 'inline';
        /**
         * Sets the color of the visual "progress bar".
         * The string should be a 6-digit hexadecimal color with a hashtag (example: #3dcd58).
         * By default, the progress bar will have a green color.
         */
        this.secolor = 'primary';
        this.isHexColor = false;
    }
    colorDidChange() {
        this.isHexColor = (this.secolor && this.secolor.indexOf('#') !== -1) ? true : false;
    }
    componentDidLoad() {
        this.colorDidChange();
    }
    render() {
        return (index.h(index.Host, { class: !this.isHexColor ? `color-${this.secolor}` : '' }, index.h("div", { class: ["se-visual-linear", this.option].join(' ') }, index.h("svg", { height: "8" }, index.h("rect", { class: "progress-bar-wrapper", width: "100%", height: "8" }), index.h("rect", { class: "progress-bar", width: this.percentage + '%', height: "8", style: { fill: this.isHexColor ? this.secolor : 'currentColor' } }), "Sorry, your browser does not support inline SVG."), index.h("text", null, this.value ? index.h("tspan", { class: "linear-value" }, this.value) : '', index.h("tspan", { x: "100%", class: "linear-label" }, this.label)))));
    }
    static get watchers() { return {
        "secolor": ["colorDidChange"]
    }; }
};
VisualLinearComponent.style = visualLinearCss;

exports.se_visual_linear = VisualLinearComponent;
