// @se/web-ui/collection/components/form-field/form-field.js
import { Component, Element, Event, h, Prop, Listen, Watch } from '@stencil/core';
export class FormFieldComponent {
    constructor() {
        /**
         * Defines the layout of your form field.
         * `inline` is the default option, and is always applied if the type is set to `checkbox`.  This sets the input or select field adjacent to the label.
         * `stacked` option will render the input or select field below the label.
         */
        this.option = 'inline';
        /**
         * Defines the spacing around the inside edge of a container.
         * `none` is 0px.
         * `small` is 4px.
         * `medium` is 8px.
         */
        this.padding = 'small';
        /**
         * Optional property to define the proportion of the label width. The content (input) will take the rest of if. Defaulted to 40%
         */
        this.labelWidth = '35%';
        /**
         * Defines whether the form field's input is a text field (`input`), a checkbox (`checkbox`), a radio button (`radio`), or a dropdown menu (`select`).
         * `input` is the default type.
         */
        this.type = 'input';
        /**
         * Determines if the input is required by the application.
         * Set to `false` by default.
         * Setting this value to `true` will render a red asterisk next to your label.
         */
        this.required = false;
        /**
         * Optional property that defines if the form field is disabled.  Set to `false` by default.
         */
        this.disabled = false;
    }
    inputSelectListenerHandler(event) {
        this.handleEvent(event);
    }
    checkboxListenerHandler(event) {
        this.handleEvent(event);
    }
    radioListenerHandler(event) {
        this.handleEvent(event);
    }
    disabledDidChange() {
        this.initLabel();
    }
    typeDidChange() {
        this.initLabel();
    }
    initLabel() {
        Array.from(this.el.querySelectorAll('input, select, textarea, se-checkbox, se-radio')).forEach((item) => {
            item.disabled = this.disabled;
        });
    }
    componentDidLoad() {
        this.initLabel();
    }
    handleEvent(event) {
        if (!this.disabled) {
            this.value = event.detail ? event.detail.selected : event.target.value;
            this.didSubmit.emit(this.value);
        }
    }
    render() {
        return (h("div", { class: {
                [`ff-${this.status}`]: true,
                'ff-stacked': this.option === 'stacked',
                'ff-block': this.block,
                [`ff-padding-${this.padding}`]: true,
                'form-field-wrapper': true
            } },
            h("label", { "data-disabled": this.disabled },
                h("span", { style: {
                        width: this.option !== 'stacked' ? this.labelWidth : 'auto'
                    }, class: { 'with-label': !!this.label } },
                    this.label,
                    this.required ? h("span", { class: "required" }, "*") : ''),
                h("div", { class: "ff-wrapper" },
                    h("div", { class: "ff-wrapper-input" },
                        h("slot", null))))));
    }
    static get is() { return "se-form-field"; }
    static get originalStyleUrls() { return {
        "$": ["form-field.scss"]
    }; }
    static get styleUrls() { return {
        "$": ["form-field.css"]
    }; }
    static get properties() { return {
        "option": {
            "type": "string",
            "mutable": false,
            "complexType": {
                "original": "'inline' | 'stacked'",
                "resolved": "\"inline\" | \"stacked\"",
                "references": {}
            },
            "required": false,
            "optional": false,
            "docs": {
                "tags": [],
                "text": "Defines the layout of your form field.\n`inline` is the default option, and is always applied if the type is set to `checkbox`.  This sets the input or select field adjacent to the label.\n`stacked` option will render the input or select field below the label."
            },
            "attribute": "option",
            "reflect": false,
            "defaultValue": "'inline'"
        },
        "padding": {
            "type": "string",
            "mutable": false,
            "complexType": {
                "original": "'none' | 'small' | 'medium'",
                "resolved": "\"medium\" | \"none\" | \"small\"",
                "references": {}
            },
            "required": false,
            "optional": false,
            "docs": {
                "tags": [],
                "text": "Defines the spacing around the inside edge of a container.\n`none` is 0px.\n`small` is 4px.\n`medium` is 8px."
            },
            "attribute": "padding",
            "reflect": false,
            "defaultValue": "'small'"
        },
        "block": {
            "type": "boolean",
            "mutable": false,
            "complexType": {
                "original": "boolean",
                "resolved": "boolean",
                "references": {}
            },
            "required": false,
            "optional": false,
            "docs": {
                "tags": [],
                "text": "Optional property that defines if the field displays as a block in it's container.\nWhen set to true, the field will be as wide as its container.\nboth field label auto adjust when it set to inline"
            },
            "attribute": "block",
            "reflect": false
        },
        "labelWidth": {
            "type": "string",
            "mutable": false,
            "complexType": {
                "original": "string",
                "resolved": "string",
                "references": {}
            },
            "required": false,
            "optional": false,
            "docs": {
                "tags": [],
                "text": "Optional property to define the proportion of the label width. The content (input) will take the rest of if. Defaulted to 40%"
            },
            "attribute": "label-width",
            "reflect": false,
            "defaultValue": "'35%'"
        },
        "type": {
            "type": "string",
            "mutable": false,
            "complexType": {
                "original": "'input' | 'checkbox' | 'radio' | 'select'",
                "resolved": "\"checkbox\" | \"input\" | \"radio\" | \"select\"",
                "references": {}
            },
            "required": false,
            "optional": false,
            "docs": {
                "tags": [],
                "text": "Defines whether the form field's input is a text field (`input`), a checkbox (`checkbox`), a radio button (`radio`), or a dropdown menu (`select`).\n`input` is the default type."
            },
            "attribute": "type",
            "reflect": false,
            "defaultValue": "'input'"
        },
        "status": {
            "type": "string",
            "mutable": true,
            "complexType": {
                "original": "'error' | 'warning' | 'success'",
                "resolved": "\"error\" | \"success\" | \"warning\"",
                "references": {}
            },
            "required": false,
            "optional": false,
            "docs": {
                "tags": [],
                "text": "Sets a red border on an input field if there's an error, an orange border if there's a warning, and a green border if a successful input."
            },
            "attribute": "status",
            "reflect": false
        },
        "label": {
            "type": "string",
            "mutable": false,
            "complexType": {
                "original": "string",
                "resolved": "string",
                "references": {}
            },
            "required": false,
            "optional": false,
            "docs": {
                "tags": [],
                "text": "Defines the text value of the label in your form field."
            },
            "attribute": "label",
            "reflect": false
        },
        "value": {
            "type": "string",
            "mutable": true,
            "complexType": {
                "original": "string",
                "resolved": "string",
                "references": {}
            },
            "required": false,
            "optional": false,
            "docs": {
                "tags": [],
                "text": "Defines the value of your form field to get passed to the parent component.\nWhen the type is set to \"input\", this value will be the default placeholder in your input field."
            },
            "attribute": "value",
            "reflect": false
        },
        "required": {
            "type": "boolean",
            "mutable": false,
            "complexType": {
                "original": "boolean",
                "resolved": "boolean",
                "references": {}
            },
            "required": false,
            "optional": false,
            "docs": {
                "tags": [],
                "text": "Determines if the input is required by the application.\nSet to `false` by default.\nSetting this value to `true` will render a red asterisk next to your label."
            },
            "attribute": "required",
            "reflect": false,
            "defaultValue": "false"
        },
        "disabled": {
            "type": "boolean",
            "mutable": false,
            "complexType": {
                "original": "boolean",
                "resolved": "boolean",
                "references": {}
            },
            "required": false,
            "optional": false,
            "docs": {
                "tags": [],
                "text": "Optional property that defines if the form field is disabled.  Set to `false` by default."
            },
            "attribute": "disabled",
            "reflect": false,
            "defaultValue": "false"
        }
    }; }
    static get events() { return [{
            "method": "didSubmit",
            "name": "didSubmit",
            "bubbles": true,
            "cancelable": true,
            "composed": true,
            "docs": {
                "tags": [],
                "text": "Passes form data to the parent component on a click (`checkbox` or `radio`), menu change (`select`), or when the input field loses focus."
            },
            "complexType": {
                "original": "any",
                "resolved": "any",
                "references": {}
            }
        }]; }
    static get elementRef() { return "el"; }
    static get watchers() { return [{
            "propName": "disabled",
            "methodName": "disabledDidChange"
        }, {
            "propName": "type",
            "methodName": "typeDidChange"
        }]; }
    static get listeners() { return [{
            "name": "change",
            "method": "inputSelectListenerHandler",
            "target": undefined,
            "capture": false,
            "passive": false
        }, {
            "name": "didChange",
            "method": "checkboxListenerHandler",
            "target": undefined,
            "capture": false,
            "passive": false
        }, {
            "name": "didCheck",
            "method": "radioListenerHandler",
            "target": undefined,
            "capture": false,
            "passive": false
        }]; }
}
